plugins/patchmanagement.koplugin/main.lua

```diff
@@ -0,0 +1,189 @@
+--[[--
+Plugin for managing user patches
+
+@module koplugin.patchmanagement
+--]]--
+
+local DataStorage = require("datastorage")
+local lfs = require("libs/libkoreader-lfs")
+
+if lfs.attributes(DataStorage:getDataDir() .. "/patches", "mode") ~= "directory" then
+    return { disabled = true }
+end
+
+local Device = require("device")
+local InfoMessage = require("ui/widget/infomessage")
+local TextViewer = require("ui/widget/textviewer")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local sort = require("sort")
+local userPatch = require("userpatch")
+local _ = require("gettext")
+local Screen = Device.screen
+
+local PatchManagement = WidgetContainer:extend{
+    name = "patch_management",
+}
+
+function PatchManagement:init()
+    self.patch_dir = DataStorage:getDataDir() .. "/patches"
+    self.disable_ext = ".disabled"
+    self.patches = nil
+    self:getAvailablePatches()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function PatchManagement:getAvailablePatches()
+    self.patches = {}
+    for priority = tonumber(userPatch.early_once), tonumber(userPatch.on_exit) do
+        self.patches[priority] = {}
+    end
+
+    for entry in lfs.dir(self.patch_dir) do
+        local mode = lfs.attributes(self.patch_dir .. "/" .. entry, "mode")
+        if mode == "file" then
+            for priority = tonumber(userPatch.early_once), tonumber(userPatch.on_exit) do
+                if entry:match("^" .. priority .. "%d*%-") and entry:find(".lua", 1, true) then
+                    -- only lua files, starting with "%d+%-"
+                    table.insert(self.patches[priority], entry)
+                    break
+                end
+            end
+        end
+    end
+
+    for priority = tonumber(userPatch.early_once), tonumber(userPatch.on_exit) do
+        table.sort(self.patches[priority], sort.natsort_cmp())
+    end
+end
+
+function PatchManagement:getSubMenu(priority)
+    if self.patches == nil then
+        return {}
+    end
+    local function getExecutionStatus(patch_name)
+        return userPatch.execution_status[patch_name] == false and " ⚠" or ""
+    end
+    local sub_menu = {}
+    for i, patch in ipairs(self.patches[priority]) do
+        local ext = ".lua"
+        -- strip anything after ".lua" in patch_name
```

Comment:
> Was this a design decision ? Or a happy side effect of overeng. ? :)

@poire-z :I think the author deliberately decided to do it hat way

Reply:
That depends on what you mean with well. 

`2-somestuff.lua.old` is handled as `2-somestuff.lua.disabled`.  So any file ending with `.luaxxx` is disabled.

So  `2-somestuff.lua.old` *will* show up in patch management as disabled. If you don't want to show it up you can name it as `x2-somestuff.lua` or `abc2-somestuff.lua` or `outdatedxxx2-somestuff.lua`.

If you have `2-somestuff.lua.old` and `2-somestuff.lua` in `patches` both are shown. The first as disabled, the second as enabled. This is not optimal, but reflects the current files.
If you dis/enable one of them, the behavior is not designed.

Reply:
So if you want to keep an old version of a patch, you can/have to rename it to `2-somestuffxxxx.lua` (I would suggest `2-somestuff-v1.lua` or `2-somestuff-20230206.lua`) and you can dis/enable it as you like.